sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py

```diff
@@ -101,18 +101,13 @@ def _translate_step_name(self, internal_name):
           'Could not translate the internal step name %r since job graph is '
           'not available.' % internal_name)
     user_step_name = None
-    # pylint: disable=wrong-import-order, wrong-import-position
-    from apache_beam.runners.dataflow.internal import apiclient
-    if apiclient._use_unified_worker_portable_job(self._job_graph.options):
+    if (self._job_graph and internal_name
+        in self._job_graph.proto_pipeline.components.transforms.keys()):
       # Dataflow Runner v2 with portable job submission uses proto transform map
       # IDs for step names. Also PTransform.unique_name maps to user step names.
       # Hence we lookup user step names based on the proto.
-      proto_pipeline = self._job_graph.proto_pipeline
-      for transform_id in proto_pipeline.components.transforms.keys():
-        if internal_name == transform_id:
-          user_step_name = proto_pipeline.components.transforms[
-              transform_id].unique_name
-          break
+      user_step_name = self._job_graph.proto_pipeline.components.transforms[
```

Comment:
Is this also valid for non-portable job submission paths? (just to make sure we don't break backwards-compatibility when users specify runner_v2, but not portable job submission).

Reply:
Yeah, we'll only enter this block if service chooses to use portable submission **and** use transform IDs as step names.